gateway-js/src/index.ts

```diff
@@ -754,8 +759,9 @@ export class ApolloGateway implements GatewayInterface {
       { attributes: spanAttributes },
       async (span) => {
         try {
-          const { request, document, queryHash } = requestContext;
-          const queryPlanStoreKey = queryHash + (request.operationName || '');
+          const { request, document } = requestContext;
+          const queryPlanKey = print(document) + (request.operationName || '');
```

Comment:
Printing a document is not super cheap, especially for large queries, so it's worth avoiding if we can. In this case, I believe the query string is available in `requestContext.request.query` so that can be used.

But as computing sha256 is not extremely cheap either and we already have the hash of the query, why not re-hash that with the operation name. Something like (happy to extract in an helper method in practice, just keeping it short here):
```typescript
const queryPlanStoreKey = request.operationName
  ? createHash('sha256').update(queryHash).update(request.operationName).digest('hex')
  : queryHash;
```

Reply:
Agreed, though maybe just hash the concatenation `queryHash + request.operationName`?

Reply:
Admittedly, I don't think this make much practical difference, but concatenating necessarily means creating a new underlying string and some memory copies, while calling `update` twice most likely doesn't. And calling `update` repeatedly is pretty idiomatic too for a hash implementation as those are usually created with streaming in mind (and [node JS seems no exception](https://nodejs.org/api/crypto.html#hashupdatedata-inputencoding)).